Updated Weather Module

DeltaOS2 opened the pull request:
More options to pick a location

DeltaOS2 commented:
I've added all options to pick a locations given by OpenWeatherMap.org

emilylaguna commented on bundled/weather.infinitymodule/package.json:
```diff
@@ -2,10 +2,12 @@
   "description": "Displays the current temperature and weather for a given location",
   "author": "FIPLAB Ltd",
-  "website": "https://infinitydashboard.com/",
+  "contributors": [
```
Can you add the website field back in? 

`"website": "https://infinitydashboard.com/",`

emilylaguna commented on bundled/weather.infinitymodule/package.json:
```diff
@@ -15,14 +17,43 @@
     {
       "title": "Location",
-      "name": "cityName",
+      "name": "loc",
```
Changing the `name` of this breaks existing installed modules when the module is updated. Please change this back to cityName to prevent this issue.

emilylaguna commented on bundled/weather.infinitymodule/main.js:
```diff
@@ -6,12 +6,34 @@ let arguments = fiplab.arguments;
+  case 'coordinates':
+    coord = loc.split(',');
+    lat = coord[0].trim();
+    lon = coord[1].trim();
```
If the user enters invalid coordinates without the `,` the module throws an exception and doesn't return a proper error. Please do some simple validation on the coordinates and  display a user friendly error message (such as: "Invalid coordinates") if it's not valid. 

![screen shot 2018-06-13 at 10 50 17 am](https://user-images.githubusercontent.com/793774/41368714-95faa632-6ef7-11e8-956d-2cf45dae1334.png)

DeltaOS2 commented:
Hi Emily,

I did all the changes you’ve asked for and updated my fork.
Do I need to pull another request?

I also updated the README.html and the README.md with examples for the location input. 
The README.md is updated for better readability, too.

I hope you’re OK with that.

Best Regards from Atlanta,
Gert Massheimer


> Am 13.06.2018 um 13:55 schrieb Emily Laguna <notifications@github.com <mailto:notifications@github.com>>:
> 
> @emilylaguna requested changes on this pull request.
> 
> Hi, thanks for submitting these changes to the default module. I've added some items for you to review before I'll merge. A couple items to note:
> 
> There's an issue when upgrading the module since one of the preferences name was changed. Since this will be bundled with the app this is not something that we can have
> If the user enters an invalid coordinate (no comma) then the module doesn't display an error. See my comments below.
> The website information was removed.
> Once these items have been updated I'll review the module again and merge if everything looks good.
> 
> In bundled/weather.infinitymodule/package.json <https://github.com/fiplab/Infinity-Dashboard-Modules/pull/4#discussion_r195168302>:
> 
> > @@ -2,10 +2,12 @@
>    "name": "Current Weather",
>    "description": "Displays the current temperature and weather for a given location",
>    "author": "FIPLAB Ltd",
> -  "website": "https://infinitydashboard.com/ <https://infinitydashboard.com/>",
> +  "contributors": [
> Can you add the website field back in?
> 
> "website": "https://infinitydashboard.com/ <https://infinitydashboard.com/>",
> 
> In bundled/weather.infinitymodule/package.json <https://github.com/fiplab/Infinity-Dashboard-Modules/pull/4#discussion_r195172254>:
> 
> >      },
>      {
>        "title": "Location",
> -      "name": "cityName",
> +      "name": "loc",
> Changing the name of this breaks existing installed modules when the module is updated. Please change this back to cityName to prevent this issue.
> 
> In bundled/weather.infinitymodule/main.js <https://github.com/fiplab/Infinity-Dashboard-Modules/pull/4#discussion_r195178145>:
> 
> > +  case 'cityName':
> +    urlFormat = 'http://api.openweathermap.org/data/2.5/weather?q=%s&units=%s&appid=%s <http://api.openweathermap.org/data/2.5/weather?q=%s&units=%s&appid=%s>';
> +    options = { 'url': util.format(urlFormat, arguments.loc, arguments.units, arguments.apiKey) }
> +    break;
> +  case 'zipCode':
> +    urlFormat = 'http://api.openweathermap.org/data/2.5/weather?zip=%s&units=%s&appid=%s <http://api.openweathermap.org/data/2.5/weather?zip=%s&units=%s&appid=%s>';
> +    options = { 'url': util.format(urlFormat, arguments.loc, arguments.units, arguments.apiKey) }
> +    break;
> +  case 'cityID': 
> +    urlFormat = 'http://api.openweathermap.org/data/2.5/weather?id=%s&units=%s&appid=%s <http://api.openweathermap.org/data/2.5/weather?id=%s&units=%s&appid=%s>';
> +    options = { 'url': util.format(urlFormat, arguments.loc, arguments.units, arguments.apiKey) }
> +    break;
> +  case 'coordinates':
> +    coord = loc.split(',');
> +    lat = coord[0].trim();
> +    lon = coord[1].trim();
> If the user enters invalid coordinates without the , the module throws an exception and doesn't return a proper error. Please do some simple validation on the coordinates and display a user friendly error message (such as: "Invalid coordinates") if it's not valid.
> 
>  <https://user-images.githubusercontent.com/793774/41368714-95faa632-6ef7-11e8-956d-2cf45dae1334.png>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub <https://github.com/fiplab/Infinity-Dashboard-Modules/pull/4#pullrequestreview-128481415>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AWXuX-fS7kc4D_FinJfILpZA_MQh8ecPks5t8VIMgaJpZM4UmPTq>.
>

emilylaguna commented:
Hi Gert, thanks for making these changes. I've gone ahead and merged your pull request. We'll be including the updated module in the next version of Infinity Dashboard.
